### DISCUSSION TITLE
PorousFlow - init stress when restarting

### DISCUSSION THREAD

**Discussion by Nuriacor** [idaholab/moose#19281] [Q&A Modules: General] [answered]
Hi,
I have a two step simulation with PorousFlow module. I first run my model in steady state and the restart the variables in a second stage to run transient conditions. As shown here (https://mooseframework.inl.gov/modules/porous_flow/restart.html)

The model is HM coupled, I'm restarting porepressure and the displacements. I notice that at the beginning (time zero) all variables are zero. Then, in the first time step, variables values are set, but the porepressure is not quite the same that in my steady state output, which generates unwanted dynamic at the beginning of my simulation.

After modifying things for a while it came to my mind that my initial stress may be set wrong. For now I have set `ComputeEigenstrainFromInitialStress` with a function that depends on density and depth (as in this tutorial https://mooseframework.inl.gov/modules/porous_flow/coal_mining.html). 

Is the initial stress messing my transient simulation? Or is it something else?

Thanks in advance.
Nuria

**Comment by WilkAndy** [comment, marked as answer]
Initial stress is a little more complicated than initial porepressure.  The first step of debugging this should be to remove the "M" coupling, and see if you can set the porepressure to *exactly* the steady-state value.  Eg, look at `porous_flow/examples/flow_through_fractured_media/fine_thick_fracture_transient.i`.  Don't simply assume everything has worked OK if the results *look* OK in paraview, since often porepressure equilibrates very quickly during the first step of your transient simulation.   Instead, actually check all the significant figures of porepressure somewhere, using, eg, a `PointValue` Postprocessor.

Then we can get on to the stresses....

a

**Reply by Nuriacor** [reply]
Hi,
Thanks for your help. I did what you suggest, I also studied the `fine_thick_fracture_transient.i` example. But I could not debug my problem.

The H coupled model works fine, so the problem may come from the HM coupling. See `test_H_steady.i `and `test_H_transient.i `files.

In the HM model, the pressure difference between the transient and the steady solution is small, but increases significantly when I set different permeability in the layers. Please see `test_HM_steady.i `and `test_HM_transient.i `files.

I also noticed that when I run the simulation in parallel the number of nonlinear and linear steps increases. At the point that is faster to run in only one core.

https://github.com/Nuriacor/porousflow_tests

Thanks again!
Nuria

**Reply by WilkAndy** [reply]
I'm up to my eyeballs in some other work, sorry for being so tardy in replying - i don't think i'll get to this today, but it seems as if something like the stress is not being set correctly, as you already know!

**Reply by WilkAndy** [reply]
Oh, i just had a quick look.  You need to set your `ini_stress_yy` from the steady-state file.  You probably don't want to set the `disp_x` and `disp_y` from the steadystate file.

**Reply by Nuriacor** [reply]
Thanks!! So do I need to output the yy stress in the steady simulation and then imported in the transient and assign it to a function?

**Reply by WilkAndy** [reply]
Yea, you need to transfer all the stress components (but prob the off-diagonal ones are all zero) into the transient simulation

**Comment by Nuriacor** [comment]
@WilkAndy I went around this with one single simulation file and using `Controls` to change the BCs. Thanks! 
I think that would be interesting to know why my simulation was behaving like that.
Cheers!

**Reply by Nuriacor** [reply]
And I am also curios to know why more cores increase the nonlinear steps.

**Reply by WilkAndy** [reply]
This is almost definitely because the linear solver is crappier.  You're choosing
```
   petsc_options_iname = '-ksp_type -pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
    petsc_options_value = 'gmres      asm      lu           NONZERO                   2             '
```
which will give the exact solution for 1 processor, but for many it'll only give an approximate solution to Ax=b.  Look at https://mooseframework.inl.gov/modules/porous_flow/solvers.html for some more choices.

**Reply by Traiwit** [reply]
Hi @WilkAndy,

i've been using `Hypre / BoomerAMG` for my full-scale hydro simulation (5m+ elements)

```

    petsc_options_iname = '-pc_type -pc_hypre_type -pc_hypre_boomeramg_strong_threshold -pc_hypre_boomeramg_agg_nl  -pc_hypre_boomeramg_agg_num_paths -pc_hypre_boomeramg_max_levels  -pc_hypre_boomeramg_coarsen_type   -pc_hypre_boomeramg_interp_type -pc_hypre_boomeramg_P_max -pc_hypre_boomeramg_truncfactor'
    petsc_options_value = 'hypre    boomeramg 0.7 4 5 25 HMIS  ext+i 2 0.3'
```
any comment on this one? I find it's the fastest so far.

**Reply by WilkAndy** [reply]
looks good to me!  i've used hypre/boomeramg with good results.
